### DISCUSSION TITLE
Delete missing ID relations

### DISCUSSION THREAD

**Discussion by dennSted** [pimcore/pimcore#17641] [Q&A / Support]
Hey everyone,

i'm not sure it this is the right place, but I hope so.
Currently i'm struggling with two different pimcore issues. One is that my log, and also when entering any command, showing multiple ERROR messages like:

`
10:06:26 ERROR     [pimcore] object with ID 486 not found
10:06:26 ERROR     [pimcore] object with ID 494 not found
10:06:26 ERROR     [pimcore] object with ID 497 not found
10:06:26 ERROR     [pimcore] object with ID 484 not found
10:06:26 ERROR     [pimcore] object with ID 504 not found
10:06:26 ERROR     [pimcore] object with ID 541 not found
10:06:26 ERROR     [pimcore] object with ID 538 not found
10:06:26 ERROR     [pimcore] object with ID 533 not found
10:06:26 ERROR     [pimcore] object with ID 416 not found
10:06:26 ERROR     [pimcore] object with ID 488 not found
10:06:26 ERROR     [pimcore] object with ID 506 not found
10:06:26 ERROR     [pimcore] object with ID 487 not found
10:06:26 ERROR     [pimcore] object with ID 505 not found
10:06:26 ERROR     [pimcore] object with ID 507 not found
10:06:26 ERROR     [pimcore] object with ID 485 not found
10:06:26 ERROR     [pimcore] object with ID 501 not found
10:06:26 ERROR     [pimcore] object with ID 544 not found
10:06:26 ERROR     [pimcore] object with ID 543 not found
10:06:26 ERROR     [pimcore] object with ID 539 not found
10:06:26 ERROR     [pimcore] object with ID 535 not found
10:06:26 ERROR     [pimcore] object with ID 75 not found
10:06:26 ERROR     [pimcore] object with ID 74 not found
`

Further when I'm trying to use the command "bin/console pimcore:deployment:classes-rebuild" I receive the following

`
10:34:39 ERROR     [pimcore] class Car doesn't exist anymore
Exception: Could not load class Car in /opt/bitnami/pimcore/vendor/pimcore/pimcore/models/DataObject/Objectbrick/Definition.php:426
Stack trace:
#0 /opt/bitnami/pimcore/vendor/pimcore/pimcore/models/DataObject/Objectbrick/Definition.php(203): Pimcore\Model\DataObject\Objectbrick\Definition->createContainerClasses()
#1 /opt/bitnami/pimcore/vendor/pimcore/pimcore/bundles/CoreBundle/src/Command/ClassesRebuildCommand.php(143): Pimcore\Model\DataObject\Objectbrick\Definition->save()
#2 /opt/bitnami/pimcore/vendor/symfony/console/Command/Command.php(326): Pimcore\Bundle\CoreBundle\Command\ClassesRebuildCommand->execute()
#3 /opt/bitnami/pimcore/vendor/symfony/console/Application.php(1096): Symfony\Component\Console\Command\Command->run()
#4 /opt/bitnami/pimcore/vendor/symfony/framework-bundle/Console/Application.php(126): Symfony\Component\Console\Application->doRunCommand()
#5 /opt/bitnami/pimcore/vendor/symfony/console/Application.php(324): Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand()
#6 /opt/bitnami/pimcore/vendor/symfony/framework-bundle/Console/Application.php(80): Symfony\Component\Console\Application->doRun()
#7 /opt/bitnami/pimcore/vendor/symfony/console/Application.php(175): Symfony\Bundle\FrameworkBundle\Console\Application->doRun()
#8 /opt/bitnami/pimcore/vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php(49): Symfony\Component\Console\Application->run()
#9 /opt/bitnami/pimcore/vendor/autoload_runtime.php(29): Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run()
#10 /opt/bitnami/pimcore/bin/console(20): require_once('...')
#11 {main}10:34:39 ERROR     [pimcore] class Car doesn't exist anymore
Exception: Could not load class Car in /opt/bitnami/pimcore/vendor/pimcore/pimcore/models/DataObject/Objectbrick/Definition.php:426
Stack trace:
#0 /opt/bitnami/pimcore/vendor/pimcore/pimcore/models/DataObject/Objectbrick/Definition.php(203): Pimcore\Model\DataObject\Objectbrick\Definition->createContainerClasses()
#1 /opt/bitnami/pimcore/vendor/pimcore/pimcore/bundles/CoreBundle/src/Command/ClassesRebuildCommand.php(143): Pimcore\Model\DataObject\Objectbrick\Definition->save()
#2 /opt/bitnami/pimcore/vendor/symfony/console/Command/Command.php(326): Pimcore\Bundle\CoreBundle\Command\ClassesRebuildCommand->execute()
#3 /opt/bitnami/pimcore/vendor/symfony/console/Application.php(1096): Symfony\Component\Console\Command\Command->run()
#4 /opt/bitnami/pimcore/vendor/symfony/framework-bundle/Console/Application.php(126): Symfony\Component\Console\Application->doRunCommand()
#5 /opt/bitnami/pimcore/vendor/symfony/console/Application.php(324): Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand()
#6 /opt/bitnami/pimcore/vendor/symfony/framework-bundle/Console/Application.php(80): Symfony\Component\Console\Application->doRun()
#7 /opt/bitnami/pimcore/vendor/symfony/console/Application.php(175): Symfony\Bundle\FrameworkBundle\Console\Application->doRun()
#8 /opt/bitnami/pimcore/vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php(49): Symfony\Component\Console\Application->run()
#9 /opt/bitnami/pimcore/vendor/autoload_runtime.php(29): Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run()
#10 /opt/bitnami/pimcore/bin/console(20): require_once('...')
#11 {main}10:34:39 ERROR     [pimcore] class Car doesn't exist anymore
Exception: Could not load class Car in /opt/bitnami/pimcore/vendor/pimcore/pimcore/models/DataObject/Objectbrick/Definition.php:426
Stack trace:
......
`
<img width="705" alt="error_pimcore_console_on_classesrebuild" src="https://github.com/user-attachments/assets/43e0a1da-26d8-454f-b517-acfc72683759">

I currently use the Demo package of pimcore but deleted some classes like CAR and AccessoryParts cause I didn't need them. After updating Pimcore I now receive these error messages and **would like to know how I can maybe delete these IDs completly for Pimcore**??

Further I hope that someone knows the reason for this weird message which I currently only receive, when trying to save or edit a specific created class:

<img width="598" alt="error_pimcore_500" src="https://github.com/user-attachments/assets/3bc94c53-d538-4342-bd02-9e06dbeb4f50">

I didn't changed anything inside this file or 
`
 /* This file is based on WebProfilerBundle/Resources/views/Profiler/base_js.html.twig.
   If you make any change in this file, verify the same change is needed in the other file. */
`
which also be marked inside the long error description.

I hope anyone knows the reason or where I have to look?

**Comment by kingjia90** [comment]
Make sure you have deleted all the existing object bricks

![image](https://github.com/user-attachments/assets/d11a755c-ecb9-4ef1-b669-37d308e69698)

They make use of the `Car` class

![image](https://github.com/user-attachments/assets/1fcc9336-bd68-44a9-b14c-686fed8a8193)

**Reply by dennSted** [reply]
Thanks for the answer. The problem is that in all still existing ObjectBricks there is no Car Class defined (as seen below in the example). Every Class is empty excepting my own created both "Product Fabric"'s where my own defined Product Class is connected.

<img width="679" alt="Information_Specs" src="https://github.com/user-attachments/assets/a02b502f-c351-4968-8bd6-ccdade29b71d">

**Reply by kingjia90** [reply]
Then i don't know, but maybe it is just not visible, but still have some usages in the DB and under the hood. M
Maybe try to manually re-save all the object bricks.
You may try to (x)debug and see which Object Brick specifically throws this exception

**Reply by dennSted** [reply]
Do you have any idea where I can maybe find some relations inside the DB? I'm currently using pimcore in debug mode and not receiving a more detailed error message or any clear info in any logfile.

Maybe do you have any idea relating my second issue https://github.com/orgs/pimcore/discussions/17641#discussioncomment-10694150:~:text=Further%20I%20hope,long%20error%20description. ?
This is one hard thing which didn't let me edit my Product class anymore manually. I can only use a workaround by Exporting the class and edit/add new fields by code and import it back. The weird thing is that I can edit other classes without any issue...
